### PR TITLE
feat(Message): member getter

### PR DESCRIPTION
**Pull request by DaStormer**
Please describe the changes this PR makes and why it should be merged:
Added member getter to message

Status

-   [x] Code changes have been tested.

Semantic versioning classification:

-   [ ] This PR changes the library's interface (methods or parameters added)
-   [ ] This PR includes breaking changes (methods removed or renamed, parameters moved or removed)
-   [ ] This PR only includes non-code changes, like changes to documentation, README, etc.

### CODE REVIEW

File: packages/guilded.js/lib/structures/Message.ts [hunk 1]
```diff
@@ -85,6 +86,11 @@ export class Message extends Base<ChatMessagePayload> {
         return this.createdByBotId ?? this.createdByWebhookId ?? this.createdById;
     }
 
+    /** Get the member of this message (if in server) */
+    get member(): Member | null {
```
Comment (zaida04) [inline packages/guilded.js/lib/structures/Message.ts:91]: Do we want to make one universal function for constructing server member cache keys?

Comment (DaStormer) [reply packages/guilded.js/lib/structures/Message.ts:91]: That would be ideal!

Comment (DaStormer) [reply packages/guilded.js/lib/structures/Message.ts:91]: Just implemented this in https://github.com/guildedjs/guilded.js/pull/75/commits/7da1b13543f246c8e600777aeb36f3646b1aba4e 😄

File: packages/guilded.js/lib/gateway/handler/TeamMemberEventHandler.ts [hunk 2]
```diff
@@ -22,13 +23,13 @@ export class TeamMemberEventHandler extends GatewayEventHandler {
...
 
-        this.client.members.cache.set(TeamMemberEventHandler.buildMemberKey(data.d.serverId, data.d.member.user.id), newMember);
+        this.client.members.cache.set(buildMemberKey(data.d.serverId, data.d.member.user.id), newMember);
         this.client.users.cache.set(newUser.id, newUser);
         return this.client.emit(constants.clientEvents.TEAM_MEMBER_JOINED, newMember);
     }
     teamMemberRemoved(data: WSTeamMemberRemovedPayload): boolean {
-        const existingMember = this.client.members.cache.get(TeamMemberEventHandler.buildMemberKey(data.d.serverId, data.d.userId));
-        if (this.client.options.cache?.removeMemberOnLeave) this.client.members.cache.delete(TeamMemberEventHandler.buildMemberKey(data.d.serverId, data.d.userId));
+        const existingMember = this.client.members.cache.get(buildMemberKey(data.d.serverId, data.d.userId));
```
Comment (zaida04) [inline packages/guilded.js/lib/gateway/handler/TeamMemberEventHandler.ts]: Can condense this line’s key and the next line’s key with a variable

File: packages/guilded.js/lib/managers/global/GuildBanManager.ts [hunk 3]
```diff
@@ -37,8 +38,8 @@ export class GlobalGuildBanManager extends CacheableStructManager<string, Member
     /** Unban a user from a server. Returns existing ban if cached. */
     unban(serverId: string, userId: string, removeBanIfCached = false): Promise<MemberBan | null> {
         return this.client.rest.router.unbanMember(serverId, userId).then((data) => {
-            const existingBan = this.client.bans.cache.get(`${serverId}:${userId}`);
-            if (this.client.options.cache?.removeMemberOnLeave || removeBanIfCached) this.client.bans.cache.delete(`${serverId}:${userId}`);
+            const existingBan = this.client.bans.cache.get(buildMemberKey(serverId, userId));
```
Comment (zaida04) [inline packages/guilded.js/lib/managers/global/GuildBanManager.ts]: Can condense this line’s key and the next line’s key with a variable

File: packages/guilded.js/lib/structures/Message.ts [hunk 4]
```diff
@@ -85,6 +87,11 @@ export class Message extends Base<ChatMessagePayload> {
...
+        return this.client.members.cache.get(buildMemberKey(this.serverId ?? "", this.authorId)) ?? null;
```
Comment (zaida04) [inline packages/guilded.js/lib/structures/Message.ts]:
```suggestion
        return this.serverId ? (this.client.members.cache.get(buildMemberKey(this.serverId, this.authorId)) ?? null) : null;
```

**Review by zaida04 — CHANGES_REQUESTED** [review]

**Review by zaida04 — APPROVED** [review]

**Pull request merged**